omegaconf/basecontainer.py

```diff
@@ -89,10 +89,30 @@ def __repr__(self) -> str:
 
     # Support pickle
     def __getstate__(self) -> Dict[str, Any]:
-        return self.__dict__
+        from omegaconf._utils import is_dict_annotation
+
+        dict_ = copy.copy(self.__dict__)
+        dict_["_metadata"] = copy.copy(dict_["_metadata"])
+        if self._has_ref_type():
+            if is_dict_annotation(self._metadata.ref_type):
+                dict_["_metadata"].ref_type = Dict
+            else:
+                dict_["_metadata"].ref_type = List
```

Comment:
defaulting to list here is bad practice. if we ever have another type this will be broken.
```python
if is_dict...:
  ...
elif is_list...:
  ...
else:
  assert False
```

Reply:
Okay